testutil/checkers.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	"fmt"
+	"gopkg.in/check.v1"
+	"reflect"
+	"strings"
+)
+
+type containsChecker struct {
+	*check.CheckerInfo
+}
+
+// Contains is a Checker that looks for a needle in a haystack.
+// The needle can be any object. The haystack can be an array, slice or string.
+var Contains check.Checker = &containsChecker{
+	&check.CheckerInfo{Name: "Contains", Params: []string{"haystack", "needle"}},
+}
+
+func (c *containsChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	defer func() {
+		if v := recover(); v != nil {
+			result = false
+			error = fmt.Sprint(v)
+		}
+	}()
+	var haystack interface{} = params[0]
+	var needle interface{} = params[1]
+	switch haystackV := reflect.ValueOf(haystack); haystackV.Kind() {
+	case reflect.Slice, reflect.Array:
+		// Ensure that type of elements in haystack is compatible with needle
+		if needleV := reflect.ValueOf(needle); haystackV.Type().Elem() != needleV.Type() {
+			panic(fmt.Sprintf("haystack contains items of type %s but needle is a %s",
```

Comment:
This should return the error in the error string, rather than being a panic. It's the only reason why the that result exists. Have a look at use cases in checkers.go in check.v1 for examples.